prow/cmd/deck/static/prow/prow.ts

```diff
@@ -571,7 +572,18 @@ function redraw(fz: FuzzySearch): void {
         r.appendChild(cell.state(state));
         if ((agent === "kubernetes" && pod_name) || agent !== "kubernetes") {
             const logIcon = icon.create("description", "Build log");
-            logIcon.href = `log?job=${job}&id=${build_id}`;
+            if (pod_spec == null || pod_spec.containers.length <= 1) {
+                logIcon.href = `log?job=${job}&id=${build_id}`;
+            } else {
+                // this logic exists for legacy jobs that are configured for gubernator compatibility
+                const buildIndex = url.indexOf('/build/');
+                if (buildIndex !== -1) {
+                    const gcsUrl = `${window.location.origin}/view/gcs/${url.substring(buildIndex + '/build/'.length)}`;
+                    logIcon.href = gcsUrl;
+                } else if (url.includes('/view/')) {
+                    logIcon.href = url;
+                }
+            }
```

Comment:
Please ensure that we always set `logIcon.href`. We should probably default to using the `log?job=${job}&id=${build_id}` format if we can't determine an appropriate spyglass link.